File: sdk/highlight-java/highlight-sdk/src/main/java/io/highlight/sdk/HighlightTracer.java
package io.highlight.sdk;

import java.time.Instant;

import io.highlight.sdk.common.HighlightAttributes;
import io.highlight.sdk.common.record.HighlightErrorRecord;
import io.opentelemetry.api.trace.Span;
import io.opentelemetry.api.trace.Tracer;
import io.opentelemetry.context.Scope;

/**
 * The HighlightTracer class is used to process error records
 */
public class HighlightTracer {

	private final Tracer tracer;

	/**
	 * Constructs a new HighlightTracer instance.
	 *
	 * @param highlight The Highlight instance.
	 */
	HighlightTracer(Highlight highlight) {
		HighlightOpenTelemetry openTelemetry = highlight.getOpenTelemetry();
		this.tracer = openTelemetry.getTracer("highlight-java");
	}

	/**
	 * Processes the given {@link HighlightErrorRecord}.
	 *
	 * @param record The HighlightErrorRecord to process.
	 */
	public void process(HighlightErrorRecord record) {
		Span span = this.tracer.spanBuilder("highlight-ctx").setAllAttributes(record.getAttributes())
				.setStartTimestamp(record.getTimeOccured()).setNoParent().startSpan();

		try (Scope scope = span.makeCurrent()) {
			if (record.hasUserSession()) {
				span.setAttribute(HighlightAttributes.HIGHLIGHT_SESSION_ID, record.getUserSession().sessionId());
			}

			if (record.hasRequestId()) {
				span.setAttribute(HighlightAttributes.HIGHLIGHT_TRACE_ID, record.getRequestId());
			}

			span.recordException(record.getThrowable());
		} finally {
			span.end(Instant.now());
		}
	}
}

File: sdk/highlight-php/src/SDK/HighlightTracer.php
<?php

declare(strict_types=1);

namespace Highlight\SDK;

use Highlight\SDK\Common\HighlightAttributes;
use Highlight\SDK\Common\Record\HighlightErrorRecord;
use OpenTelemetry\API\Trace\TracerInterface;
use OpenTelemetry\Context\Context;

class HighlightTracer
{

    private TracerInterface $tracer;

    public function __construct(Highlight $highlight)
    {
        $highlightOpenTelemetry = $highlight->getOpenTelemetry();
		$this->tracer = $highlightOpenTelemetry->getTracerProvider()->getTracer('highlight-php'); 
    }

    public function process(HighlightErrorRecord $record)
    {
        $spanBuilder = $this->tracer->spanBuilder('highlight-ctx');

        $spanBuilder->setAttributes($record->getAttributes()->toArray())
            ->setStartTimestamp($record->getTimeOccurred()->getTimestamp())
            ->setParent(null);

        $span = $spanBuilder->startSpan();

        try {
            Context::storage()->attach($span->storeInContext(Context::getCurrent()));

            if ($record->hasUserSession()) {
                $span->setAttribute(HighlightAttributes::HIGHLIGHT_SESSION_ID, $record->getUserSession()->sessionId());
            }

            if ($record->hasRequestId()) {
                $span->setAttribute(HighlightAttributes::HIGHLIGHT_TRACE_ID, $record->getRequestId());
            }

            $span->recordException($record->getThrowable());

            // You can add additional logic here to handle exceptions, set status codes, and more.

        } finally {
            $span->end();
        }
    }
}
